src/compat/compat.h

```diff
@@ -607,6 +607,23 @@ pg_strtoint64(const char *str)
 							  freezeLimit,                                                         \
 							  multiXactCutoff);                                                    \
 	} while (0)
+#else
+#define vacuum_set_xid_limits_compat(rel,                                                          \
+									 freeze_min_age,                                               \
+									 freeze_table_age,                                             \
+									 multixact_freeze_min_age,                                     \
+									 multixact_freeze_table_age,                                   \
+									 oldestXmin,                                                   \
+									 freezeLimit,                                                  \
+									 multiXactCutoff)                                              \
+	do                                                                                             \
+	{                                                                                              \
+		struct VacuumCutoffs cutoffs;                                                              \
+		vacuum_get_cutoffs(OldHeap, 0, &cutoffs);                                                  \
```

Comment:
Should `OldHeap` be `rel` here?
Also, is the current implementation of `vacuum_get_cutoffs` happy with a `NULL` `params` pointer?

Reply:
> Should `OldHeap` be `rel` here? 

Yes, my bad. I have changed it now.

>Also, is the current implementation of `vacuum_get_cutoffs` happy with a `NULL` `params` pointer?

I checked - it doesn't handle NULL params. So, I have updated the patch to send a empty params object rather than a NULL.